Fix crash in winregistry.py

laxa opened the pull request:
I encountered the following crash when using `registry-read.py`:

```
$ registry-read.py registry__system.hiv enum_values -name 'ControlSet001\Services\LanmanServer\Parameters'
Impacket v0.10.0 - Copyright 2022 SecureAuth Corporation

[ControlSet001\Services\LanmanServer\Parameters]

[b'EnableAuthenticateUserSharing', b'NullSessionPipes', b'ServiceDll', b'ServiceDllUnloadOnStop', b'autodisconnect', b'enableforcedlogoff', b'enablesecuritysignature', b'requiresecuritysignature', b'restrictnullsessaccess', b'Guid']
  b'EnableAuthenticateUserSharing':  0
  b'NullSessionPipes'           :  Traceback (most recent call last):
  File "/home/laxa/.local/bin/registry-read.py", line 172, in <module>
    main()
  File "/home/laxa/.local/bin/registry-read.py", line 161, in main
    enumValues(reg, options.name)
  File "/home/laxa/.local/bin/registry-read.py", line 97, in enumValues
    reg.printValue(data[0],data[1])
  File "/home/laxa/.local/lib/python3.9/site-packages/impacket/winregistry.py", line 410, in printValue
    print("%s" % (valueData.decode('utf-16le')))
AttributeError: 'int' object has no attribute 'decode'
```

This patch is straightforward and fix the aforementioned issue.

anadrianmanrique commented:
question: which was the value of the NullSessionPipes key? was it 0? it had no value?

anadrianmanrique commented:
0 is not the actual value for that key, as it was empty. I provided feedback in the code review section in order to simplify and fix this issue. thanks!

laxa commented:
Hello, I am not sure where I should access the code review section, I cannot access any comments from the menu available in the PR, could you provide a link to this section? Thanks.

laxa commented:
I cannot see any comments related to the code, I don't know if its related to rights access.

anadrianmanrique commented on impacket/winregistry.py:
```diff
@@ -407,7 +407,10 @@ def printValue(self, valueType, valueData):
                 print(" NULL")
         elif valueType == REG_MULTISZ:
-            print("%s" % (valueData.decode('utf-16le')))
+            if isinstance(valueData, int):
```
sorry, my bad, check now

anadrianmanrique commented:
sorry, my bad, check now

laxa commented:
All good, I can see your comment properly. I made the change and switched the `type(valueData)` to `isinstance` because it offers better performances. I also squashed the first commit.

anadrianmanrique commented:
Great! thanks for the PR fix
